app/controllers/application_controller.rb

```diff
@@ -94,7 +94,11 @@ def scoped_sign_in_path
     if request.path =~ /\/cfp/
       new_cfp_session_path
     else
-      new_session_path
+      if request.get?
+        new_session_path(:return_to => request.path)
```

Comment:
`return_to: request.path` looks much nicer and is also what the code uses elsewhere.